spyder/widgets/collectionseditor.py

```diff
@@ -644,99 +642,146 @@ def setup_table(self):
+            name=None,
+            text=_("Insert below"),
             icon=ima.icon('insert_below'),
-            triggered=lambda: self.insert_item(below=True)
+            triggered=lambda: self.insert_item(below=True),
+            register_action=False
         )
-        self.remove_action = create_action(self, _("Remove"),
-                                           icon=ima.icon('editdelete'),
-                                           triggered=self.remove_item)
-        self.rename_action = create_action(self, _("Rename"),
-                                           icon=ima.icon('rename'),
-                                           triggered=self.rename_item)
-        self.duplicate_action = create_action(self, _("Duplicate"),
-                                              icon=ima.icon('edit_add'),
-                                              triggered=self.duplicate_item)
-        self.view_action = create_action(
-            self,
-            _("View with the Object Explorer"),
+        self.remove_action = self.create_action(
+            name=None,
+            text=_("Remove"),
+            icon=ima.icon('editdelete'),
+            triggered=self.remove_item,
+            register_action=False
+        )
+        self.rename_action = self.create_action(
+            name=None,
+            text=_("Rename"),
+            icon=ima.icon('rename'),
+            triggered=self.rename_item,
+            register_action=False
+        )
+        self.duplicate_action = self.create_action(
+            name=None,
+            text=_("Duplicate"),
+            icon=ima.icon('edit_add'),
+            triggered=self.duplicate_item,
+            register_action=False
+        )
+        self.view_action = self.create_action(
+            name=None,
+            text=_("View with the Object Explorer"),
             icon=ima.icon('outline_explorer'),
-            triggered=self.view_item)
+            triggered=self.view_item,
+            register_action=False
+        )
 
-        menu = SpyderMenu(self)
-        self.menu_actions = [
-            self.edit_action,
+        menu = self.create_menu('Editor menu', register=False)
+        menu_actions = [
             self.copy_action,
             self.paste_action,
             self.rename_action,
-            self.remove_action,
+            self.edit_action,
             self.save_array_action,
             MENU_SEPARATOR,
             self.insert_action,
             self.insert_action_above,
             self.insert_action_below,
             self.duplicate_action,
+            self.remove_action,
             MENU_SEPARATOR,
             self.view_action,
             self.plot_action,
             self.hist_action,
-            self.imshow_action,
-            MENU_SEPARATOR,
-            resize_action,
-            resize_columns_action
+            self.imshow_action
         ]
-        add_actions(menu, self.menu_actions)
+        add_actions(menu, menu_actions)
```

Comment:
This menu should also be built as explained above, i.e. by declaring sections and then using `add_item_to_menu`.